src/main/kotlin/com/autonomousapps/model/GradleVariantIdentification.kt

```diff
@@ -39,4 +43,39 @@ data class GradleVariantIdentification(
       target.gradleVariantIdentification.capabilities.containsAll(capabilities)
     }
   }
+
+  companion object {
+    val EMPTY = GradleVariantIdentification(emptySet(), emptyMap())
+  }
+}
+
+/**
+ * KMP artifacts have a [KotlinPlatformType] attribute defined for which target it is.
+ *
+ * At a general level, these can be thought of as a canonical proxy dependency and "the rest", which
+ * covers specific platform targets like [KotlinPlatformType.jvm], [KotlinPlatformType.native], etc.
+ *
+ * When inspecting attributes for the context of DAGP, we want to know if a given dependency is a canonical
+ * dependency or a specific target. When recommending advice, we want to defer to the canonical deps as
+ * a sort of implicit bundle of its other targets and will resolve to the correct target.
+ */
+internal val Coordinates.kmpAttribute get() = gradleVariantIdentification.attributes[KotlinPlatformType.attribute.name]
+
+/**
+ * Returns whether this is a canonical KMP dependency, where "canonical" means it is not an alias to an external
+ * variant that it's been resolved to.
+ */
+internal val Coordinates.isKmpCanonicalDependency: Boolean
+  get() = kmpAttribute != null && gradleVariantIdentification.externalVariant != null
+
+/** Returns whether this is a specific target KMP dependency, such as [KotlinPlatformType.jvm]. */
+internal val Coordinates.isKmpTargetTarget: Boolean
+  get() = kmpAttribute != null && gradleVariantIdentification.externalVariant == null
+
+/**
+ * Returns a [Coordinates.gav] string that represents the parent [Coordinates.identifier] of the given KMP target's
+ * `common` parent.
+ */
+internal fun ModuleCoordinates.kmpCommonParentIdentifier(): String {
+  return identifier.substringBeforeLast('-')
```

Comment:
How safe is this string manipulation? Do we "know" that they'll always use a suffix that starts with a `-`?

Reply:
as far as I can tell, yes. I'd love a more programmatic way but I haven't been able to decipher where in Gradle's APIs one could find this information and think it's just implicit.